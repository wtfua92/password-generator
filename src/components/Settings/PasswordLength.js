import React from "react";
import PropTypes from "prop-types";

function PasswordLength({ passwordLength, changePasswordLengthHandler }) {
  return (
    <div className="password-generator__settings__length">
      <label htmlFor="passwordLengthNumber">Length: </label>
      <input
        id="passwordLength"
        type="range"
        min={8}
        max={16}
        value={passwordLength}
        onChange={({ target }) =>
          changePasswordLengthHandler(parseInt(target.value))
        }
      />
      <input
        id="passwordLengthNumber"
        type="number"
        min={8}
        max={16}
        value={passwordLength}
        onChange={({ target }) =>
          changePasswordLengthHandler(parseInt(target.value))
        }
      />
    </div>
  );
}

PasswordLength.propTypes = {
  passwordLength: PropTypes.number.isRequired,
  changePasswordLengthHandler: PropTypes.func.isRequired
};

PasswordLength.defaultProps = {
  passwordLength: 8,
  changePasswordLengthHandler: () => {}
};

export default PasswordLength;
