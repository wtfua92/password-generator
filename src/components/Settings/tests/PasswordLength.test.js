import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import PasswordLength from "../PasswordLength";

describe("<PasswordLength />", function() {
  const changePasswordLengthHandlerMock = jest.fn();
  const props = {
    passwordLength: 12,
    changePasswordLengthHandler: changePasswordLengthHandlerMock
  };

  let container;

  beforeEach(() => {
    container = render(<PasswordLength {...props} />);
    changePasswordLengthHandlerMock.mockReset();
  });

  afterEach(cleanup);

  it("should display both range and number inputs", () => {
    const { getAllByDisplayValue } = container;
    const inputs = getAllByDisplayValue(props.passwordLength.toString());
    expect(inputs.length).toEqual(2);
  });

  it("should call changePasswordLengthHandler on change event of input[type='number']", () => {
    const { getAllByDisplayValue } = container;
    const input = getAllByDisplayValue(props.passwordLength.toString()).find(
      i => i.type === "number"
    );
    fireEvent.change(input, { target: { value: 13 } });
    expect(changePasswordLengthHandlerMock).toHaveBeenCalledTimes(1);
    expect(changePasswordLengthHandlerMock).toHaveBeenCalledWith(13);
  });

  it("should call changePasswordLengthHandler on change event of input[type='range']", () => {
    const { getAllByDisplayValue } = container;
    const input = getAllByDisplayValue(props.passwordLength.toString()).find(
      i => i.type === "range"
    );
    fireEvent.change(input, { target: { value: 13 } });
    expect(changePasswordLengthHandlerMock).toHaveBeenCalledTimes(1);
    expect(changePasswordLengthHandlerMock).toHaveBeenCalledWith(13);
  });

  it("should match snapshot", () => {
    const {
      container: { firstChild: root }
    } = container;
    expect(root).toMatchSnapshot();
  });
});
