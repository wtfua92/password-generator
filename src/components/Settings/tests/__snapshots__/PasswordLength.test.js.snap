// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordLength /> should match snapshot 1`] = `
<div
  class="password-generator__settings__length"
>
  <label
    for="passwordLengthNumber"
  >
    Length: 
  </label>
  <input
    id="passwordLength"
    max="16"
    min="8"
    type="range"
    value="12"
  />
  <input
    id="passwordLengthNumber"
    max="16"
    min="8"
    type="number"
    value="12"
  />
</div>
`;
